docs/en/observability/apm/apm-performance-diagnostic.asciidoc

```diff
@@ -0,0 +1,43 @@
+{"log.level":"error","@timestamp":"2024-07-27T23:55:38.612Z","log.origin":{"function":"github.com/elastic/go-docappender/v2.(*Appender).flush","file.name":"v2@v2.2.0/appender.go","file.line":370},"message":"bulk indexing request failed","service.name":"apm-server","error":{"message":"flush failed (503): [503 Service Unavailable]"},"ecs.version":"1.6.0"}
+...
+----
+
+To gain better insight into APM Server health and performance, consider enabling the monitoring feature by following the steps in <<apm-monitor-apm-self-install,Monitor a Fleet-managed APM Server>>.
+When enabled APM Server will additionally report a set of vital metrics to help you identify any performance degradation.
+
+Pay careful attention to the next metric fields:
+
+* `beats_stats.metrics.libbeat.output.events.active` that represents the number of buffered pending documents waiting for indexing;
+(_if this value is increasing rapidly it indicates {es} backpressure_)
+* `beats_stats.metrics.libbeat.output.events.acked` that represents the number of indexing operations that have completed successfully;
+* `beats_stats.metrics.libbeat.output.events.failed` that represents the number of indexing operations that failed, it includes all failures;
+(_if this value is increasing rapidly it indicates {es} backpressure_)
+* `beats_stats.metrics.libbeat.output.events.toomany` that represents the number of indexing operations that failed due to {es} responding with 429 Too many Requests;
+(_if this value is increasing rapidly it indicates {es} backpressure_)
+* `beats_stats.output.elasticsearch.bulk_requests.available` that represents the number of bulk indexers available for making bulk index requests;
+(_if this value is equal to 0 it indicates {es} backpressure_)
+* `beats_stats.output.elasticsearch.bulk_requests.completed` that represents the number of already completed bulk requests;
+* `beats_stats.metrics.output.elasticsearch.indexers.active` that represents the number of active bulk indexers that are concurrently processing batches;
+
+See https://www.elastic.co/guide/en/beats/metricbeat/current/exported-fields-beat.html[{metricbeat} documentation] for the full list of exported metric fields.
+
+One likely cause of excessive indexing pressure or rejected requests is undersized {es}. To mitigate this, follow the guidance in {ref}/rejected-requests.html[Rejected requests].
+If scaling {es} resources up is not an option, you can try to workaround by adjusting `flush_bytes`, `flush_interval`, `max_retries` and `timeout` settings described in <<apm-elasticsearch-output,Configure the Elasticsearch output>> to reduce APM Server indexing pressure.
```

Comment:
```suggestion
If scaling {es} resources up is not an option, you can adjust the `flush_bytes`, `flush_interval`, `max_retries` and `timeout` settings described in <<apm-elasticsearch-output,Configure the Elasticsearch output>> to reduce APM Server indexing pressure.
```

Should we also include suggestions on how to tune the values?
Or give a sort of reference config? Not sure if we can provide one that is valid for all customers though

Reply:
Added an example yaml config. As you said it's not applicable to specific situation but probably it's still nice to have a config reference example in the doc in front of you.

Reply:
Tbh I am not a big fan of providing an example here. If I were the user I would be tempted to skip the advice about scaling up ES and conveniently apply this example config and continue to get a bad experience from our APM offering. But now that it is added to the PR, I don't feel strongly to remove it, as we clearly stated that "if scaling ES is not an option".

Reply:
I can see your point Carson, and how that might be tempting to just copy paste the config example instead of more direct and effective scale up of ES.  I feel like it's a matter of preference whether to introduce this section or not, since it has value of quickly providing a visual config reference but at the same time it might encourage users to skip reading other parts of this doc.

We probably should be consistent with the other documentation pages, do we usually add config example in other places in the documentation?

Also an alternative approach is to keep the example, but move it under a subsection "Workaround by configuring APM" at the bottom of the page where in the first sentence we state that it's not recommended to apply it, and only try to modify configs if the other options are exhausted.

Reply:
sounds good, left a suggestion.